### PR TITLE
apply license gradle plugin

### PR DESCRIPTION
**Pull request by sdelamo**
This PR applies the [Gradle License Plugin](https://github.com/hierynomus/license-gradle-plugin). The plugin contributes a gradle task `downloadLicenses` which generates a license report of every dependencies.

The plugin is applied to subprojects and the root project to get an aggregated report.

```
./gradlew downloadLicenses
open build/reports/license/license-dependency.html
```

### CODE REVIEW

**Review by matrei — COMMENTED** [review]
To prevent bloating the root build file, could we extract this to a new file, for example: `gradle/dependency-licenses.gradle`?

**Comment by sdelamo** [comment]
> To prevent bloating the root build file, could we extract this to a new file, for example: `gradle/dependency-licenses.gradle`?

@matrei I did that in https://github.com/grails/grails-core/pull/13535/commits/9e7b15316cfa3a891850e9de8039ac9dd0698eba

File: gradle/dependency-licenses.gradle [hunk 1]
```diff
@@ -0,0 +1,55 @@
+buildscript {
+    repositories {
+        maven { url "https://plugins.gradle.org/m2/" }
+    }
+    dependencies {
+
+        classpath("gradle.plugin.com.hierynomus.gradle.plugins:license-gradle-plugin:0.16.1")
+    }
+}
```
Comment (matrei) [inline gradle/dependency-licenses.gradle]: Is it allowed to have `buildScript` blocks in included files?
I think, as this is already present in `build.gradle` it is not needed here.

File: build.gradle [hunk 2]
```diff
@@ -290,6 +294,7 @@ allprojects {
 }
 
 apply from: "gradle/idea.gradle"
+apply from: "gradle/dependency-licenses.gradle"
```
Comment (matrei) [inline build.gradle]: Should the `dependency-licenses.gradle` file be applied here, even though it is also applied on line 623?

File: build.gradle [hunk 3]
```diff
@@ -352,6 +357,7 @@ subprojects { project ->
     apply plugin: 'eclipse'
     apply plugin: 'idea'
     apply plugin: 'project-report'
+    apply plugin: 'com.github.hierynomus.license-report'
```
Comment (matrei) [inline build.gradle]: Should the plugin be applied directly here, even though you extracted the functionality to `dependency-licenses.gradle` file and applied it on line 623?

**Review by matrei — CHANGES_REQUESTED** [review]
Nice, now just one more thing I saw (I'm a bit allergic to complex build files):
As we already have an `allprojects` block in `build.gradle`, we can apply the `gradle/dependency-licenses.gradle` file there, so we don't have to do it in multiple places:
```groovy
allprojects {
    
    apply from: rootProject.layout.projectDirectory.file('gradle/dependency-licenses.gradle')

    ...
}
```

File: gradle/dependency-licenses.gradle [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+        "org.grails:grails-test",
+        "org.grails:grails-test-suite-base",
+        "org.grails:grails-test-suite-persistence",
+        "org.grails:grails-test-suite-uber",
+        "org.grails:grails-test-suite-web",
+        "org.grails:grails-web",
+        "org.grails:grails-web-boot",
+        "org.grails:grails-web-common",
+        "org.grails:grails-web-databinding",
+        "org.grails:grails-web-fileupload",
```
Comment (matrei) [inline gradle/dependency-licenses.gradle]: Oh, and `org.grails:grails-web-fileupload` is removed in 7.0.x.

If all we want is to exclude all the subprojects, could we do this instead to not have to maintain the list:
```groovy
// We don't want to include any of the libraries from this repo in the license report
List<String> licenseExclusions = rootProject.subprojects.collect {
    "org.grails:${it.name}:${rootProject.projectVersion}" as String
}
```

and we probably want to be able to verify the list:

```groovy
tasks.register('printLicenseExclusions') {
    group = 'license'
    description = 'Prints the license exclusions list'
    doLast {
        println """
            |============================================================
            |The following projects are excluded from the license report:
            |============================================================
            |${licenseExclusions.join('\n')}
        """.stripMargin()
    }
}
```

**Review by matrei — CHANGES_REQUESTED** [review]

**Comment by matrei** [comment]
@sdelamo Sorry, for the delay in reviewing, I had not fully understood the process so my comments were left in pending state.

**Review by matrei — APPROVED** [review]

**Pull request merged**